utilities/copyoffload_migration.py

```
@@ -1349,6 +1351,33 @@ def _verify_source_host_labels_on_pods(populate_pods: list[Pod]) -> str:
+
+    Returns:
+        str: The shared sourceHost label value.
+
+    Raises:
+        ValueError: If source_host is missing from any entry or hosts are inconsistent across pods.
+    """
+    source_hosts: set[str] = set()
+    for log_data in pod_logs:
+        source_host = log_data["source_host"]
+        if not source_host:
+            raise ValueError(f"Populate pod '{log_data['pod_name']}' is missing cached '{SOURCE_HOST_LABEL}' label")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)